File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        unordered_set<int> st;
        for(int &it : nums){
            st.insert(it);
        }
        int longest = 0;
        for(int i=0;i<nums.size();i++){
            if(st.find(nums[i]-1) != st.end()){
                continue;
            }
            else{
                int count =0;
                int current = nums[i];
                while(st.find(current) != st.end()){
                    count++;
                    current++;
                }
                longest = max(count,longest);
            }
        }
        return longest;
    }
};

File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.py
class Solution(object):
    def longestConsecutive(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        if(len(nums) == 0):
            return 0
        nums.sort()
        count = 1
        largest =1
        i=1
        while(i<len(nums)):
            if(nums[i-1]+1 == nums[i]):
                count +=1
            else:
                largest = max(count,largest)
                count =1
            i +=1
        
        largest = max(count,largest)
        
        return largest
        
        